mitodl/mit-open
# Fix poetry lockfile, update generate_openapi formattting

# What are the relevant tickets?
None. The last heroku deploy failed.

# Description (What does it do?)
This PR:
1. Updates `poetry.lock` in an attempt to fix the heroku deployment. The [last deploy](https://dashboard.heroku.com/apps/mitopen-rc/activity/builds/43251a64-f611-4a40-8d02-a4b044b2b96f) failed due to python installation problems. 
2. Updates `ci.yaml` to check that `poetry.lock` is consistent with `pyproject.toml`
3. Updates `generate_openapi.sh` to format generated code using pre-commit commands.

# How can this be tested?
1. If you want, run `docker compose exec web  poetry lock --check` to check that the lockfile is consistent, but CI is doing this also. https://github.com/mitodl/mit-open/actions/runs/6148820905/job/16683385187?pr=74
2. Run `./scripts/generate_openapi.sh` and check the logs...
    - openapi.yaml and `/frontends/api/src/generated` should be re-generated and formatted
3. Maybe `pre-commit run --all-files` since there's an issue running that on CI right now.
